frontend/src/helpers/renderHelpers.tsx

```diff
@@ -1,18 +1,26 @@
 import { SectionBox } from '../components/common';
+import { KubeObject } from '../lib/k8s/cluster';
 import { useTypedSelector } from '../redux/reducers/reducers';
 
-export default function DetailsViewPluginRenderer(props: { resource: any }) {
+function prepareDetailsComponent(
+  Component: (props: { resource: KubeObject }) => JSX.Element | null,
+  resource: KubeObject
+) {
+  return <Component resource={resource} />;
+}
+
+export default function DetailsViewPluginRenderer(props: { resource: KubeObject }) {
   const { resource } = props;
   const detailViews = useTypedSelector(state => state.ui.views.details.pluginAppendedDetailViews);
   return (
     <>
       {detailViews.map((item, index) => {
         const pluginDetailsObj = item.sectionFunc(resource);
         if (pluginDetailsObj) {
-          const { title, component: DetailsView } = pluginDetailsObj;
+          const { title } = pluginDetailsObj;
           return (
             <SectionBox title={title} key={`${item.sectionName}__${index}`}>
-              <DetailsView resource={resource} />
+              {prepareDetailsComponent(pluginDetailsObj.component, resource)}
```

Comment:
Could this be something like this instead? (to avoid having to make the prepareDetailsComponent function)
```tsx
          const { title, component: Component } = pluginDetailsObj;
          return (
            <SectionBox title={title} key={`${item.sectionName}__${index}`}>
              <Component resource={resource} />
          ...
```